### PR TITLE
Update Contributing Guidelines

### LINKED ISSUE #66: Contributing Guidelines
Write contributing guidelines

### PR DESCRIPTION
**Pull request by Snugug**
Resolves #66

**CHANGELOG**

* :memo: Large update to Contributing guidelines

### CODE REVIEW

**Comment by Snugug** [comment]
Passing tests blocked on #170

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,11 +1,83 @@
...
+  * `feature/` - update to or new general feature not covered by other prefixes
+
+## Creating a New Version
+
+Versioning is done through [SEMVER](http://semver.org/). When creating a new version, issue a [pull request](#pull-requests) from `develop` into `master` and create new release branch off of `master` with the version's name, and create a new tag with `v` prefixed with the version's name from that branch. 
+
+For instance, if you are creating version `1.1.0`, you would start by merging `develop` into `master`, create a branch `release/1.1.0` from `master`, and create a tag `v1.1.0` from branch `release/1.1.0`.
+
+## Emoji Cheatsheet
+
```
Comment (ryanbrownhill) [inline CONTRIBUTING.md]: YES!! I approve! :+1:

Comment (Snugug) [inline CONTRIBUTING.md]: What about for the Emoji? You approve of the ones selected? Check out the [full Emoji cheatsheet](http://www.emoji-cheat-sheet.com/) if you wanna pick other ones

Comment (ryanbrownhill) [inline CONTRIBUTING.md]: I approve of the ones selected - I have the [Emoji Guide wiki page](https://github.com/IBM-Watson/design-library/wiki/Emoji-Guide) which has two additional emojis for issues I don't know if we want to include those as well.

**Comment by Snugug** [comment]
@ryanbrownhill There isn't an emoji for adding code. Should there be? Or is that too much?

**Comment by ryanbrownhill** [comment]
@Snugug I was thinking the same thing - when I was making commits a second ago maybe.

:new: `:new:` - when adding new code

**Comment by Snugug** [comment]
I'm not sure if I like `:new:`. I think maybe it'll clutter PRs too much when people are adding a lot of code.

**Comment by ryanbrownhill** [comment]
Ok well I guess a PR means you are most likely adding code anyways. The other emojis are acting for quicker filtering when looking at commit messages. Adding code doesn't necessarily need to be called out with an Emoji.

**Comment by Snugug** [comment]
Agreed.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,11 +1,91 @@
...
+* Do not squash or rebase your commits when submitting a Pull Request. It makes it much harder to follow your work and make incremental changes.
+* Update the [CHANGELOG](#maintaining-thechangelog) with your changes.
+* Ensure no Emoji tags are used in the title of your Pull Request
+
+### Git Commit Messages
+
+* Use the present tense (`"Add feature"` not `"Added Feature"`)
+* Use the imperative mood (`"Move cursor to…"` not `"Moves cursor to…"`)
+* Limit the first line to 72 characters or less
+* Include relevant Emoji from our [Emoji cheatsheet](#emoji-cheatsheet)
```
Comment (ryanbrownhill) [inline CONTRIBUTING.md]: Maybe change to: "Include relevant Emoji at the **start** of your commit." You mention it below in the emoji section but not here.

**Comment by Snugug** [comment]
If these look good to @IBM-Watson/pattern-library-owners, someone can merge it in! If you'd like to read the whole thing, compiled, [take a look](https://github.com/Snugug/patterns/blob/226a968d20b42636247459608d6a124e040b014b/CONTRIBUTING.md)

**Pull request merged**